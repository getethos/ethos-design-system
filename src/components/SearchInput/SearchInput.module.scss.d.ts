// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'SearchIcon': string;
  'SearchIconCompact': string;
  'SearchInputCompact': string;
  'SearchInputContainer': string;
}
export const cssExports: CssExports;
export default cssExports;
