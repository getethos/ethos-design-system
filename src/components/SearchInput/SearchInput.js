import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styles from './SearchInput.module.scss'
import textInputStyles from '../TextInput/TextInput.module.scss'

/**
 * SearchInput is a primitive input with a search magnifying glass
 * icon to the left. It's meant to be used as a simple primitive
 * along with `AsyncTypeahead` (likely what you want to go look at)
 *
 * See also:
 * @see src/components/AsyncTypeahead/AsyncTypeahead.js
 *
 * @public
 *
 * @param {object} props -
 * @param {boolean} disabled - whether to disable or not
 * @param {string} data-tid - required test id for the component
 * @param {string} name - required name of component
 * @param {string} value - value
 * @param {function} onBlur - onBlur callback
 * @param {function} onFocus - onFocus callback
 * @param {function} onKeyDown - onKeyDown callback
 * @param {function} onChange - onChange callback
 * @param {string} placeholder - placeholder text
 *
 * @return {JSX.Element}
 */
export const SearchInput = ({
  disabled = false,
  name,
  value,
  onBlur,
  onFocus,
  onKeyDown,
  onChange,
  placeholder = 'Search',
  ...rest
}) => {
  const [lastValue, setLastValue] = useState('')
  const handleOnChange = (ev) => {
    const val = ev.target.value
    console.log('SearchInput --> handleOnChange: ', val)
    setLastValue(val)
    if (onChange) {
      onChange(ev)
    }
  }

  return (
    <div className={styles.SearchInputContainer}>
      <input
        type="text"
        disabled={disabled}
        onBlur={onBlur}
        onChange={handleOnChange}
        onFocus={onFocus}
        onKeyDown={onKeyDown}
        className={textInputStyles.TextInput}
        value={value || lastValue}
        data-tid={rest['data-tid']}
        placeholder={placeholder}
        name={name}
      />
      <FontAwesomeIcon className={styles.SearchIcon} icon={['far', 'search']} />
    </div>
  )
}

SearchInput.propTypes = {
  'data-tid': PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  placeholder: PropTypes.string,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
}
