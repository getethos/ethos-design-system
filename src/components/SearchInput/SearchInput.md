```
const enterHandler = (ev) => {
  alert("got called!") 
}
<SearchInput onEnter={enterHandler} data-tid="search-input-tid" labelCopy="Search Input" />
```
