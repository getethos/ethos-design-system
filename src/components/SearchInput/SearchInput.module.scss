.SearchInputContainer {
  flex: 0 0 auto;
  position: relative;
  /** We need to ensure things like the async typeahead dropdown options
   * have a lower z-index so we don't inadvertantly get the very first
   * item hovered when it's half way tucked behind the input
   */
  z-index: 4;

  /* TextInput Overrides */
  > input {
    height: 40px;
    border-radius: 3px;
    /* TODO -- this is overriding EDS TextInput's base padding */
    padding: 17px var(--Space-16) 15px var(--Space-40);
    /* TODO -- use base nora-font-size or something */
    font-size: 15px;
  }
}

.SearchIcon {
  flex: 0 0 auto;
  height: var(--Space-14);
  /* Looks like FA does the fill="currentColor" trick */
  /* so we use the font color (not fill) */
  color: var(--GraySecondary--translucent);
  left: var(--Space-16);
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  transform: translateY(-50%);
}
