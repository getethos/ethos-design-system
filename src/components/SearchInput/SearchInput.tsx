import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as React from 'react'
import { useState } from 'react'
import { PublicFormProps } from '../Form/Form'
import textInputStyles from '../TextInput/TextInput.module.scss'
import styles from './SearchInput.module.scss'
type SearchInputProps = PublicFormProps & {
  value?: string
  compact?: boolean
  disabled?: boolean
  placeholder?: string
  onFocus?: (...args: any[]) => any
  onClick?: (...args: any[]) => any
  onBlur?: (...args: any[]) => any
  onChange?: (...args: any[]) => any
  onKeyDown?: (...args: any[]) => any
}
/**
 * SearchInput is a primitive input with a search magnifying glass
 * icon to the left. It's meant to be used as a simple primitive
 * along with `AsyncTypeahead` (likely what you want to go look at)
 *
 * See also:
 * @see src/components/AsyncTypeahead/AsyncTypeahead.js
 *
 * @public
 *
 * @return {JSX.Element}
 */
export const SearchInput: React.SFC<SearchInputProps> = ({
  disabled = false,
  compact = false,
  name,
  value,
  onBlur,
  onFocus,
  onClick,
  onKeyDown,
  onChange,
  placeholder = 'Search',
  ...rest
}) => {
  const [lastValue, setLastValue] = useState('')
  const handleOnChange = (ev) => {
    const val = ev.target.value
    setLastValue(val)
    if (onChange) {
      onChange(ev)
    }
  }
  const getContainerClasses = () => {
    let classes = styles.SearchInputContainer
    if (compact) {
      classes = `${styles.SearchInputContainer} ${styles.SearchInputCompact}`
    }
    return classes
  }
  return (
    <div className={getContainerClasses()}>
      <input
        type="text"
        disabled={disabled}
        onBlur={onBlur}
        onChange={handleOnChange}
        onFocus={onFocus}
        onClick={onClick}
        onKeyDown={onKeyDown}
        className={textInputStyles.TextInput}
        value={value || lastValue}
        data-tid={rest['data-tid']}
        placeholder={placeholder}
        name={name}
      />
      <FontAwesomeIcon
        className={compact ? styles.SearchIconCompact : styles.SearchIcon}
        icon={['far', 'search']}
      />
    </div>
  )
}
