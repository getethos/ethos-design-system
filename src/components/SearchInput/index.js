export { SearchInput } from './SearchInput.js'
