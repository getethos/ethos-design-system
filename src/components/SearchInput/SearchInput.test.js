import React from 'react'
import renderer from 'react-test-renderer'
import { SearchInput } from './SearchInput'

jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: '',
}))

describe('SearchInput', () => {
  it('default rendering', () => {
    const enterHandlerStub = jest.fn()
    const tree = renderer
      .create(
        <SearchInput
          onEnter={enterHandlerStub}
          data-tid="search-input-tid"
          name="search-input"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
