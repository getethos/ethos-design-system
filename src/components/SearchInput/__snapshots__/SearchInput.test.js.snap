// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput default rendering 1`] = `
<div
  className="SearchIconContainer"
>
  <input
    className="TextInput"
    data-tid="search-input-tid"
    disabled={false}
    name="search-input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <
    className="SearchIcon"
    icon={
      Array [
        "far",
        "search",
      ]
    }
  />
</div>
`;
